File: product/book.py
from product.base import Product
from notifier import SHIPPING_LABEL


class BookProduct(Product):
    @property
    def has_shipping(self):
        return True

    @property
    def price(self):
        return self._data['price']

    @property
    def notification(self):
        return 'Book product doesn\'t has to pay tax because of Constituição Art. 150, VI, d', SHIPPING_LABEL

File: backend-challenge/src/main/kotlin/domain/product/Book.kt
package domain.product

import domain.payment.ProductType
import domain.shipment.BookShipment
import domain.shipment.ShipmentStrategy
import domain.shipment.ShippingLabel

data class Book(
        val name: String,
        val type: ProductType,
        val price: Double,
        val shippingLabel: ShippingLabel? = null
) : Product() {
    override fun shipmentStrategy(): ShipmentStrategy = BookShipment()
}